Geometry_Engine/Modify/MergeVertices.cs

```diff
@@ -34,104 +39,78 @@ public static partial class Modify
         /**** Public Methods                            ****/
         /***************************************************/
 
-        public static Mesh MergedVertices(this Mesh mesh, double tolerance = Tolerance.Distance) //TODO: use the point matrix 
+        [PreviousVersion("6.0", "BH.Engine.Geometry.Modify.MergedVertices(BH.oM.Geometry.Mesh, System.Double)")]
+        [Description("Merges duplicate vertices of the mesh and ensures faces are updated to have their indecies pointig at the index of the new merged vertex.")]
+        [Input("mesh", "The mesh to merge duplicate vertices of.")]
+        [Input("tolerance", "The maximum allowable distance between two vertices for them to be deemed the same vertex.", typeof(Length))]
+        [Output("mesh", "Mesh with merged vertices.")]
+        public static Mesh MergeVertices(this Mesh mesh, double tolerance = Tolerance.Distance)
         {
-            List<Face> faces = mesh.Faces.Select(x => x.DeepClone()).ToList();
-            List<VertexIndex> vertices = mesh.Vertices.Select((x, i) => new VertexIndex(x.DeepClone(), i)).ToList();
-            
-            foreach( Face face in faces)
-            {
-                vertices[face.A].Faces.Add(face);
-                vertices[face.B].Faces.Add(face);
-                vertices[face.C].Faces.Add(face);
+            if (mesh == null)
+                return null;
 
-                if (face.IsQuad())
-                    vertices[face.A].Faces.Add(face);
-            }
+            //Set up list on structs containing location and index
+            List<Tuple<Point,int>> vertices = mesh.Vertices.Select((x, i) => new Tuple<Point, int>(x, i)).ToList();
```

Comment:
Just curious, cant we switch to tuples written as `(Point, int)`?

Reply:
possibly. Not sure what version of c# that requires though, compared to which version we officially support.